okio/src/jvmMain/kotlin/okio/ResourceFileSystem.kt

```diff
@@ -105,7 +105,9 @@ class ResourceFileSystem internal constructor(
     val existing = jarCache[this]
     if (existing != null) return existing
 
-    val created = SYSTEM.openZip(this)
+    val created = okio.internal.openZip(this, SYSTEM) { zipEntry ->
```

Comment:
I wonder whether this is clearer with the filter as a named Param. It's not obvious to mean that a filter is the defining block here. Your call.

Reply:
Done!